homeassistant/components/mysensors.py

```diff
@@ -0,0 +1,219 @@
+"""
+homeassistant.components.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+MySensors component that connects to a MySensors gateway via pymysensors
+API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.mysensors.html
+
+
+New features:
+
+New MySensors component.
+Updated MySensors Sensor platform.
+New MySensors Switch platform.
+Multiple gateways are now supported.
+
+Configuration.yaml:
+
+mysensors:
+  port:
+    - '/dev/ttyUSB0'
+    - '/dev/ttyACM1'
+  debug: true
+  persistence: true
+  persistence_file:
+    - 'path/to/.homeassistant/mysensors.json'
+    - 'path/to/.homeassistant/mysensors2.json'
+  version: '1.5'
+
+sensor:
+  platform: mysensors
+
+switch:
+  platform: mysensors
+"""
+import logging
+
+from homeassistant.helpers import (validate_config)
+
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+    TEMP_CELCIUS)
+
+CONF_PORT = 'port'
+CONF_DEBUG = 'debug'
+CONF_PERSISTENCE = 'persistence'
+CONF_PERSISTENCE_FILE = 'persistence_file'
+CONF_VERSION = 'version'
+
+DOMAIN = 'mysensors'
+DEPENDENCIES = []
+REQUIREMENTS = [
+    'https://github.com/theolind/pymysensors/archive/'
+    '2aa8f32908e8c5bb3e5c77c5851db778f8635792.zip#pymysensors==0.3']
+_LOGGER = logging.getLogger(__name__)
+ATTR_PORT = 'port'
+ATTR_DEVICES = 'devices'
+ATTR_NODE_ID = 'node_id'
+ATTR_CHILD_ID = 'child_id'
+ATTR_UPDATE_TYPE = 'update_type'
+
+IS_METRIC = None
+CONST = None
+GATEWAYS = None
+EVENT_MYSENSORS_NODE_UPDATE = 'MYSENSORS_NODE_UPDATE'
+
+
+def setup(hass, config):  # noqa
+    """ Setup the MySensors component. """
+    # pylint:disable=no-name-in-module
+    import mysensors.mysensors as mysensors
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_PORT]},
+                           _LOGGER):
+        return False
+
+    version = config[DOMAIN].get(CONF_VERSION, '1.4')
+
+    global CONST
+    if version == '1.4':
+        import mysensors.const_14 as const
+        CONST = const
+    elif version == '1.5':
+        import mysensors.const_15 as const
+        CONST = const
+    else:
+        import mysensors.const_14 as const
+        CONST = const
+
+    # Just assume celcius means that the user wants metric for now.
+    # It may make more sense to make this a global config option in the future.
+    global IS_METRIC
+    IS_METRIC = (hass.config.temperature_unit == TEMP_CELCIUS)
```

Comment:
The sensors themselves are supposed to report in either imperial or metric though, based on the configuration of the gateway.  The constructor for the Gateway doesn't have a parameter to change the value of the metric member, but it is modified to be whatever HA is configured to use: https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/sensor/mysensors.py#L87, which looks like where this code was adapted from.

Defaulting to metric shouldn't be a big deal, since like you said HA will convert for you, but in theory, it could break sensors that only report in imperial (although you could argue that's the fault of whoever built the sensor).

There are also other sensor types that could use imperial measurements (I don't use any of them personally) that this change would affect, and HA does not automatically convert those - a distance sensor, for example.